lib/ansible/plugins/inventory/vmware_vm_inventory.py

```diff
@@ -112,10 +112,7 @@
 from ansible.plugins.inventory import BaseInventoryPlugin, Cacheable
 
 
-class InventoryModule(BaseInventoryPlugin, Cacheable):
-
-    NAME = 'vmware_vm_inventory'
-
+class BaseVMwareInventory(BaseInventoryPlugin, Cacheable):
```

Comment:
not sure why you are splitting a base class, this should not be geared for others subclassing the base from the plugin itself, if you really want a base class it should be moved to live with the others.

Reply:
vmware_vm_inventory and vmware_host_inventory shares code base so I wanted to have common code as part of BaseVMwareInventory and the host and guest inventory will subclass and override them.

Reply:
@bcoca I moved all common code to a base class and will reuse it in vmware_host_inventory.